electrumsv/daemon.py

```diff
@@ -146,14 +150,55 @@ def __init__(self, fd, is_gui: bool) -> None:
             app_state.fx = FxTask(app_state.config, self.network)
             self.fx_task = app_state.async_.spawn(app_state.fx.refresh_loop)
         self.wallets = {}
-        # Setup JSONRPC server
+
+        # RPC API - (synchronous) - self.run()
         self.init_server(config, fd, is_gui)
-        # self.init_thread_watcher()
+
+        # REST API - (asynchronous) - async_ loop/thread
+        self.init_restapi_server(config, fd)
+        self.default_endpoints = {"/"    : self.status,
+                                  "/ping": self.rest_ping}
+        app_state.dapp_extensions_added = False
```

Comment:
Why do we need this attribute? Between this and the initialisation in the request loop, it seems like a better solution is needed.

Reply:
Yes. Definitely sub-optimal. I have an idea. Will see what you think next commit.

Reply:
I think this is fixed now. I made use of the config.get("cmd") and config.get("daemon_app_module") to see if it is a daemon or a daemon+dapp and if it has a dapp then wait until setup has completed --> then enter main loop of daemon.

For gui, I have disabled restapi.